### PR TITLE
feat: use ts-pattern to simplify conditions

### PR DESCRIPTION
**Pull request by raviteja83**
- Sidebar and misc refactor: https://100ms.atlassian.net/browse/WEB-2676

### CODE REVIEW

File: packages/roomkit-react/src/Prebuilt/components/MoreSettings/SplitComponents/DesktopOptions.tsx [hunk 1]
```diff
@@ -145,41 +146,46 @@ export const DesktopOptions = ({
...
+                  </Flex>
+                </Dropdown.Item>
+              );
+            })
+            .otherwise(() => (
               <Dropdown.Item
-                onClick={() => hmsActions.setAppData(APP_DATA.hlsStats, !enablHlsStats)}
-                data-testid="hls_stats"
+                onClick={() => updateState(MODALS.STATS_FOR_NERDS, true)}
+                data-testid="stats_for_nreds_btn"
```
Comment (KaustubhKumar05) [inline packages/roomkit-react/src/Prebuilt/components/MoreSettings/SplitComponents/DesktopOptions.tsx:181]: typo in testid

File: packages/roomkit-react/src/Prebuilt/components/Polls/Voting/QuestionCard.jsx [hunk 2]
```diff
@@ -92,20 +93,30 @@ export const QuestionCard = ({
...
           css={{
-            color:
-              respondedToQuiz && !isLive
-                ? isCorrectAnswer
-                  ? '$alert_success'
-                  : '$alert_error_default'
-                : '$on_surface_low',
+            color: match({ respondedToQuiz, isLive, isCorrectAnswer })
+              .when(
+                ({ respondedToQuiz, isLive }) => respondedToQuiz && !isLive,
```
Comment (KaustubhKumar05) [inline packages/roomkit-react/src/Prebuilt/components/Polls/Voting/QuestionCard.jsx:98]: This returns a boolean for color?

Comment (KaustubhKumar05) [reply packages/roomkit-react/src/Prebuilt/components/Polls/Voting/QuestionCard.jsx:98]: Or is this the nesting pattern

File: packages/roomkit-react/src/Prebuilt/components/Polls/Voting/QuestionCard.jsx [hunk 3]
```diff
@@ -92,20 +93,30 @@ export const QuestionCard = ({
...
             alignItems: 'center',
             gap: '$4',
           }}
         >
-          {respondedToQuiz && isCorrectAnswer && pollEnded ? <CheckCircleIcon height={16} width={16} /> : null}
-          {respondedToQuiz && !isCorrectAnswer && pollEnded ? <CrossCircleIcon height={16} width={16} /> : null}
+          {match({ respondedToQuiz, pollEnded, isCorrectAnswer })
+            .when(
+              ({ respondedToQuiz, pollEnded }) => respondedToQuiz && pollEnded,
+              ({ isCorrectAnswer }) => {
```
Comment (KaustubhKumar05) [inline packages/roomkit-react/src/Prebuilt/components/Polls/Voting/QuestionCard.jsx:111]: What does L110 do?

File: packages/roomkit-react/src/Prebuilt/provider/roomLayoutProvider/index.tsx [hunk 4]
```diff
@@ -21,7 +21,7 @@ export const RoomLayoutContext = React.createContext<
 // The default merge in lodash merges the values of current layout and the changes.
 // This behaviour makes changes in array based values inconsistent since a union happens.
 // The customizer uses the new value provided if one of the values is an array
-function customizer(objValue: any, srcValue: any) {
+function customizer(objValue: unknown, srcValue: unknown) {
```
Comment (KaustubhKumar05) [inline packages/roomkit-react/src/Prebuilt/provider/roomLayoutProvider/index.tsx:24]: Is there a functional difference?

Comment (raviteja83) [reply packages/roomkit-react/src/Prebuilt/provider/roomLayoutProvider/index.tsx:24]: no any gives lint warning

File: packages/roomkit-react/src/Prebuilt/components/Polls/Voting/QuestionCard.jsx (hunk 2 above)
Comment (raviteja83) [reply packages/roomkit-react/src/Prebuilt/components/Polls/Voting/QuestionCard.jsx:98]: whatever is returned in the last callback fn that is the value

**Review by KaustubhKumar05 — APPROVED** [review]

**Pull request merged**